Build on termux fails, fix hopefully inside

### What happened?

I tried to build 86box in termux again, it failed for me at linking stage with duplicated symbols

Adding patch like this

```
git diff
diff --git a/src/thread.cpp b/src/thread.cpp
index f2a0ceaf0..30ca19726 100644
--- a/src/thread.cpp
+++ b/src/thread.cpp
@@ -5,6 +5,8 @@
 #include <86box/plat.h>
 #include <86box/thread.h>

+#if !defined(__unix__)
+
 struct event_cpp11_t {
     std::condition_variable cond;
     std::mutex              mutex;
@@ -135,3 +137,5 @@ thread_destroy_event(event_t *handle)
     delete event;
 }
 }
+
+#endif
\ No newline at end of file

```

fixed build for me.

resulting 86box starts ok with default config, I can interact with IBM PC Basic ;)


### Configuration file

```ini
[General]
update_icons = 0
host_cpu = Unknown

[Machine]
machine = ibmpc
cpu_family = 8088
cpu_speed = 4772728
cpu_multi = 1
cpu_use_dynarec = 0
time_sync = local
fpu_softfloat = 0
mem_size = 64

[Video]
gfxcard = mda

[Input devices]
mouse_type = none

[Sound]
fm_driver = nuked

[Network]
net_01_link = 0
net_02_link = 0
net_03_link = 0
net_04_link = 0

[Storage controllers]
hdc = none
cassette_mode = load
```


### Operating system

Android 11

### CPU

Cortex-A55

### 86Box version

d67d86d2668b57f940620bb4b16e64e255b57cbc

### Build architecture

Linux - ARM (64-bit)

### Build type

- [ ] New recompiler
- [ ] Debug build

### Download source

Official website (Jenkins, GitHub)

### Additional context

_No response_

well new git (with different fix). still fails

```
ld.lld: error: duplicate symbol: thread_create_named
>>> defined at thread.cpp:18
>>>            CMakeFiles/86Box.dir/thread.cpp.o:(thread_create_named)
>>> defined at unix_thread.c:36
>>>            unix/CMakeFiles/plat.dir/unix_thread.c.o:(.text+0x2C)

ld.lld: error: duplicate symbol: thread_wait
>>> defined at thread.cpp:29
>>>            CMakeFiles/86Box.dir/thread.cpp.o:(thread_wait)
>>> defined at unix_thread.c:51
>>>            unix/CMakeFiles/plat.dir/unix_thread.c.o:(.text+0x98)

ld.lld: error: duplicate symbol: thread_create_event
>>> defined at thread.cpp:84
>>>            CMakeFiles/86Box.dir/thread.cpp.o:(thread_create_event)
>>> defined at unix_thread.c:56
>>>            unix/CMakeFiles/plat.dir/unix_thread.c.o:(.text+0xA4)

ld.lld: error: duplicate symbol: thread_set_event
>>> defined at thread.cpp:114
>>>            CMakeFiles/86Box.dir/thread.cpp.o:(thread_set_event)
>>> defined at unix_thread.c:68
>>>            unix/CMakeFiles/plat.dir/unix_thread.c.o:(.text+0xE4)

ld.lld: error: duplicate symbol: thread_reset_event
>>> defined at thread.cpp:125
>>>            CMakeFiles/86Box.dir/thread.cpp.o:(thread_reset_event)
>>> defined at unix_thread.c:79
>>>            unix/CMakeFiles/plat.dir/unix_thread.c.o:(.text+0x11C)

ld.lld: error: duplicate symbol: thread_wait_event
>>> defined at thread.cpp:91
>>>            CMakeFiles/86Box.dir/thread.cpp.o:(thread_wait_event)
>>> defined at unix_thread.c:89
>>>            unix/CMakeFiles/plat.dir/unix_thread.c.o:(.text+0x148)

ld.lld: error: duplicate symbol: thread_destroy_event                                              >>> defined at thread.cpp:135
>>>            CMakeFiles/86Box.dir/thread.cpp.o:(thread_destroy_event)
>>> defined at unix_thread.c:114
>>>            unix/CMakeFiles/plat.dir/unix_thread.c.o:(.text+0x22C)

ld.lld: error: duplicate symbol: thread_create_mutex                                               >>> defined at thread.cpp:38
>>>            CMakeFiles/86Box.dir/thread.cpp.o:(thread_create_mutex)
>>> defined at unix_thread.c:125
>>>            unix/CMakeFiles/plat.dir/unix_thread.c.o:(.text+0x258)

ld.lld: error: duplicate symbol: thread_wait_mutex
>>> defined at thread.cpp:56
>>>            CMakeFiles/86Box.dir/thread.cpp.o:(thread_wait_mutex)
>>> defined at unix_thread.c:136
>>>            unix/CMakeFiles/plat.dir/unix_thread.c.o:(.text+0x288)

ld.lld: error: duplicate symbol: thread_test_mutex
>>> defined at thread.cpp:46
>>>            CMakeFiles/86Box.dir/thread.cpp.o:(thread_test_mutex)
>>> defined at unix_thread.c:146
>>>            unix/CMakeFiles/plat.dir/unix_thread.c.o:(.text+0x2A8)

ld.lld: error: duplicate symbol: thread_release_mutex
>>> defined at thread.cpp:67
>>>            CMakeFiles/86Box.dir/thread.cpp.o:(thread_release_mutex)
>>> defined at unix_thread.c:156
>>>            unix/CMakeFiles/plat.dir/unix_thread.c.o:(.text+0x2C8)

ld.lld: error: duplicate symbol: thread_close_mutex
>>> defined at thread.cpp:79
>>>            CMakeFiles/86Box.dir/thread.cpp.o:(thread_close_mutex)
>>> defined at unix_thread.c:165
>>>            unix/CMakeFiles/plat.dir/unix_thread.c.o:(.text+0x2E8)
c++: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/CMakeFiles/86Box.dir/build.make:1857: src/86Box] Error 1
make[1]: *** [CMakeFiles/Makefile2:525: src/CMakeFiles/86Box.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

01267965ca1694862834da7cf5c125512bb70d5c

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -131,7 +131,6 @@ option(RTMIDI       "RtMidi"                                                    
 option(FLUIDSYNTH   "FluidSynth"                                                    ON)
 option(MUNT         "MUNT"                                                          ON)
 option(VNC          "VNC renderer"                                                  OFF)
-option(CPPTHREADS   "C++11 threads"                                                 ON)
 option(NEW_DYNAREC  "Use the PCem v15 (\"new\") dynamic recompiler"                 OFF)
 option(MINITRACE    "Enable Chrome tracing using the modified minitrace library"    OFF)
 option(GDBSTUB      "Enable GDB stub server for debugging"                          OFF)
@@ -141,8 +140,10 @@ option(DEBUGREGS486 "Enable debug register opeartion on 486+ CPUs"              
 
 if(WIN32)
     set(QT ON)
+    option(CPPTHREADS "C++11 threads" OFF)
 else()
     option(QT "Qt GUI" ON)
+    option(CPPTHREADS "C++11 threads" ON)
 endif()
 
 # Development branch features

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -27,9 +27,7 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
     add_compile_definitions(_FILE_OFFSET_BITS=64 _LARGEFILE_SOURCE=1 _LARGEFILE64_SOURCE=1)
 endif()
 
-if(WIN32)
-    target_sources(86Box PRIVATE qt/win_thread.c)
-else()
+if(CPPTHREADS)
     target_sources(86Box PRIVATE thread.cpp)
 endif()
 

--- a/src/qt/CMakeLists.txt
+++ b/src/qt/CMakeLists.txt
@@ -227,6 +227,10 @@ if(WIN32 AND NOT MINGW)
     target_sources(plat PRIVATE win_opendir.c)
 endif()
 
+if(WIN32 AND NOT CPPTHREADS)
+    target_sources(plat PRIVATE win_thread.c)
+endif()
+
 if(WIN32)
     target_sources(plat PRIVATE win_serial_passthrough.c win_netsocket.c)
 else()

--- a/src/thread.cpp
+++ b/src/thread.cpp
@@ -5,7 +5,7 @@
 #include <86box/plat.h>
 #include <86box/thread.h>
 
-struct event_cpp11_ex_t {
+struct event_cpp11_t {
     std::condition_variable cond;
     std::mutex              mutex;
     bool                    state = false;
@@ -82,14 +82,14 @@ thread_close_mutex(mutex_t *_mutex)
 event_t *
 thread_create_event()
 {
-    auto ev = new event_cpp11_ex_t;
+    auto ev = new event_cpp11_t;
     return ev;
 }
 
 int
 thread_wait_event(event_t *handle, int timeout)
 {
-    auto event = reinterpret_cast<event_cpp11_ex_t *>(handle);
+    auto event = reinterpret_cast<event_cpp11_t *>(handle);
     auto lock  = std::unique_lock<std::mutex>(event->mutex);
 
     if (timeout < 0) {
@@ -112,7 +112,7 @@ thread_wait_event(event_t *handle, int timeout)
 void
 thread_set_event(event_t *handle)
 {
-    auto event = reinterpret_cast<event_cpp11_ex_t *>(handle);
+    auto event = reinterpret_cast<event_cpp11_t *>(handle);
     {
         auto lock    = std::unique_lock<std::mutex>(event->mutex);
         event->state = true;
@@ -123,7 +123,7 @@ thread_set_event(event_t *handle)
 void
 thread_reset_event(event_t *handle)
 {
-    auto event   = reinterpret_cast<event_cpp11_ex_t *>(handle);
+    auto event   = reinterpret_cast<event_cpp11_t *>(handle);
     auto lock    = std::unique_lock<std::mutex>(event->mutex);
     event->state = false;
 }
@@ -131,7 +131,7 @@ thread_reset_event(event_t *handle)
 void
 thread_destroy_event(event_t *handle)
 {
-    auto event = reinterpret_cast<event_cpp11_ex_t *>(handle);
+    auto event = reinterpret_cast<event_cpp11_t *>(handle);
     delete event;
 }
 }

